app/src/main/java/com/example/mensfashion/core/base/BaseFragment.kt

```diff
@@ -6,18 +6,28 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.databinding.ViewDataBinding
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import androidx.navigation.NavController
+import androidx.navigation.fragment.findNavController
 import com.example.mensfashion.App
 
-abstract class BaseFragment<bindingObj : ViewDataBinding> : Fragment() {
+abstract class BaseFragment<bindingObj : ViewDataBinding,vm:ViewModel> : Fragment() {
```

Comment:
Sometimes we don't need a view model 
And in the application  of the "interface segregation principle," we put the minimum requirement for the user in the interface
please find 
https://www.linkedin.com/posts/ismail-elmogy_solidabrprinciples-programming-developer-activity-6896016407520497664-D5Eb/